### DISCUSSION TITLE
How to align tick chart with bar chart on temporal data?

### DISCUSSION THREAD

**Discussion by yuntan** [vega/altair#2590] [Q&A] [answered]
```python
import numpy as np
import pandas as pd
import altair as alt

df = pd.DataFrame(dict(
    x=pd.date_range('2022-04-01', '2022-04-10', freq='1d', inclusive='both'),
    y1=np.arange(10),
    y2=np.arange(10) + 5,
    tick_y=np.arange(10) + 7,
))

bar_chart = alt.Chart(df).mark_bar().encode(
    x=alt.X(
        'utcyearmonthdate(x):T',
        title='date',
        axis=alt.Axis(format='%m/%d', labelAlign='left')
    ),
    y=alt.Y('y1:Q', title='Y'),
    y2='y2:Q',
    tooltip=['x', 'y1', 'y2', 'tick_y'],
)

tick_chart = alt.Chart(df).mark_tick(
    thickness=2,
).encode(
    x=alt.X(
        'utcyearmonthdate(x):T',
    ),
    y='tick_y:Q',
)

bar_chart + tick_chart
```

expected: ticks are aligned with bars, and ticks and bars have the same width.

actual:

<img width="469" alt="スクリーンショット 2022-04-14 7 38 53" src="https://user-images.githubusercontent.com/574575/163281519-0e5f4c08-3a15-46fe-aa09-ca734f29f4d3.png">

**Comment by jakevdp** [comment, marked as answer]
One easy way to make this happen is to use ordinal encodings for the x-axis:
```
    x=alt.X(
        'utcyearmonthdate(x):O',
    ),
```

**Reply by yuntan** [reply]
Wow! This figure is what I expected! Thank you!

<img width="253" alt="スクリーンショット 2022-04-14 8 22 52" src="https://user-images.githubusercontent.com/574575/163285489-f5e2f93d-fdcb-4283-adb1-6fa27c2d940f.png">

**Reply by RmStorm** [reply]
But the problem with this approach is that if you have missing days they simply disappear. By using the `:T` classification, days without data show up correctly but the bars are always left-aligned.

I would really like to use `:T` instead of  `:O` and still center align the bars. I've been trying for a while without luck now so if one you knows how to do it I would love to know!

**Reply by jonmmease** [reply]
Cross reference this Vega-Lite proposal, which aims to add this ability: https://github.com/vega/vega-lite/issues/9025

**Reply by RmStorm** [reply]
Thanks for the very quick response Jon! And yeah that's exactly it! In that issue Nicolas points out all the problems with different options I've tried and some I hadn't though of. I guess I will track that issue and hope it goes somewhere. For now I will just stick with left aligned bars.

**Reply by a3agalyan** [reply]
Hey guys, what version of altair you using? I couldn't replicate the same behavior for ":O" in 5.2.0.
